### ISSUE TITLE
Overriding default Managed Identify Credential for service bus triggered functions

### ISSUE THREAD

**Issue by ChuckJonas** [Azure/azure-functions-host#8646]
#### Is your question related to a specific version? If so, please specify:

Latest

#### What language does your question apply to? (e.g. C#, JavaScript, Java, All) 

ALL

#### Question

We are trying to run a Service Bus triggered function locally on an Azure VM with access provided by the developers AD identity.

```
{
  "bindings": [
    {
      "topicName": "sb-test",
      "subscriptionName": "sb-test",
      "connection": "SB",
      "name": "message",
      "type": "serviceBusTrigger",
      "direction": "in"
    }
  ],
  "scriptFile": "../dist/servicebus-ad-trigger-example/index.js"
}
```

```
#env
 "SB__fullyQualifiedNamespace": "our.servicebus.windows.net",
```

According to the [docs on local development with managed identity](https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference?tabs=blob#local-development-with-identity-based-connections), the azure credential is obtained searching for connections in the following order:

- A local cache shared between Microsoft applications
- The current user context in Visual Studio
- The current user context in Visual Studio Code
- The current user context in the Azure CLI

The problem we are running into is that our **Azure VM's also have a Managed Identity and that ends up being what the connection** is made with.   The VM's are not configured with access to the Service Bus. 

We need to instead use the AD identify of the individuals, which would come from one of the following locations of the above  "credential search".

Surely there must be some way to override this default order?  We unfortunately don't have access to the Azure Directory Service Principle configuration or can change how the VM's are configured  (it's controlled another team).

The docs do state you can override the above logic using:

`Tenant ID`, `Client ID` & `Client secret`, but why do we actually need to provide the AD Service Principle client information?   If a `<CONNECTION_NAME_PREFIX>__tenantId` matches one of the tenants found in the above "credential search", isn't that enough to continue?  

AFIAK, we don't have the ability to provide each developer with a `Client ID` & `Client secret`...

**Comment by pjirsa** [comment]
@ChuckJonas as you noted, the DefaultAzureCredential is searching for a valid credential in a specified order. If it finds an environment or managedidentity credential, it will use those first.

WORKAROUND SOLUTION: I recently ran into this scenario and discovered that if I set the `AZURE_CLIENT_ID` env variable (using env or local settings file) to **ANY** value, it will break the lookup chain and default to using current user context from Visual Studio or VS Code.

Kudos to @nicholasfaiello for discovering this workaround.
